sixtyfps_compiler/generator/cpp.rs

```diff
@@ -1413,6 +1413,34 @@ fn access_named_reference(
     access_member(&nr.element(), nr.name(), component, component_cpp)
 }
 
+/// Returns the code that can access the component of the given element
+fn access_element_component<'a>(
+    element: &ElementRc,
+    current_component: &Rc<Component>,
+    component_cpp: &'a str,
+) -> Cow<'a, str> {
+    let e = element.borrow();
+    let enclosing_component = e.enclosing_component.upgrade().unwrap();
+    if Rc::ptr_eq(current_component, &enclosing_component) {
+        component_cpp.into()
+    } else {
+        access_element_component(
+            element,
+            &current_component
+                .parent_element
+                .upgrade()
+                .unwrap()
+                .borrow()
+                .enclosing_component
+                .upgrade()
+                .unwrap(),
+            &format!("{}->parent", component_cpp),
+        )
+        .to_string()
+        .into()
+    }
```

Comment:
This doesn't *have* to be a recursion, I could also write it as a simple loop. Do you have a preference?

Reply:
i don't have a preference.